src/panels/lovelace/editor/card-editor/hui-card-layout-editor.ts

```diff
@@ -0,0 +1,266 @@
+    ) => ({
+      rows:
+        options?.grid_rows ??
+        defaultOptions?.grid_rows ??
+        DEFAULT_GRID_OPTIONS.grid_rows,
+      columns:
+        options?.grid_columns ??
+        defaultOptions?.grid_columns ??
+        DEFAULT_GRID_OPTIONS.grid_columns,
+    })
+  );
+
+  private _isDefault = memoizeOne(
+    (options?: LovelaceLayoutOptions) =>
+      options?.grid_columns === undefined && options?.grid_rows === undefined
+  );
+
+  render() {
+    return html`
+      <div class="header">
+        <p class="intro">
+          ${this.hass.localize(
+            `ui.panel.lovelace.editor.edit_card.layout.explanation`
+          )}
+        </p>
+        <ha-button-menu
+          slot="icons"
+          @action=${this._handleAction}
+          @click=${preventDefault}
+          @closed=${stopPropagation}
+          fixed
+          .corner=${"BOTTOM_END"}
+          .menuCorner=${"END"}
+        >
+          <ha-icon-button
+            slot="trigger"
+            .label=${this.hass.localize("ui.common.menu")}
+            .path=${mdiDotsVertical}
+          >
+          </ha-icon-button>
+
+          <ha-list-item graphic="icon" .disabled=${!this._uiAvailable}>
+            ${this.hass.localize("ui.panel.lovelace.editor.edit_card.edit_ui")}
+            ${!this._yamlMode
+              ? html`
+                  <ha-svg-icon
+                    class="selected_menu_item"
+                    slot="graphic"
+                    .path=${mdiCheck}
+                  ></ha-svg-icon>
+                `
+              : nothing}
+          </ha-list-item>
+
+          <ha-list-item graphic="icon">
+            ${this.hass.localize(
+              "ui.panel.lovelace.editor.edit_card.edit_yaml"
+            )}
+            ${this._yamlMode
+              ? html`
+                  <ha-svg-icon
+                    class="selected_menu_item"
+                    slot="graphic"
+                    .path=${mdiCheck}
+                  ></ha-svg-icon>
+                `
+              : nothing}
+          </ha-list-item>
+        </ha-button-menu>
+      </div>
+      ${this._yamlMode
+        ? html`
+            <ha-yaml-editor
+              .hass=${this.hass}
+              .defaultValue=${this.config.layout_options}
+              @value-changed=${this._valueChanged}
+            ></ha-yaml-editor>
+          `
+        : html`
+            <ha-grid-size-picker
+              .hass=${this.hass}
+              .value=${this._gridSizeValue(
+                this.config.layout_options,
+                this._defaultLayoutOptions
+              )}
+              .isDefault=${this._isDefault(this.config.layout_options)}
+              @value-changed=${this._gridSizeChanged}
+            ></ha-grid-size-picker>
+          `}
+    `;
+  }
+
+  protected firstUpdated(changedProps: PropertyValues<this>): void {
+    super.firstUpdated(changedProps);
+    try {
+      this._cardElement = document.createElement("hui-card");
```

Comment:
Instead of creating another card element, can we hook into the preview element we already display?

Reply:
Not really because we need the card updated event and it's already caught by the section component. But maybe it should work because the editor will be loaded after the card. I will do some other tests! 🙂